python/ray/actor.py

```diff
@@ -125,7 +125,7 @@ def __call__(self, *args, **kwargs):
     def remote(self, *args, **kwargs):
         return self._remote(args, kwargs)
 
-    def _remote(self, args, kwargs, num_return_vals=None):
+    def _remote(self, args=None, kwargs=None, num_return_vals=None):
```

Comment:
If this works, then great, but you might need to add the following

```python
args = [] if args is None else args
kwargs = {} if kwargs is None else kwargs
```
since the downstream code is probably expecting a list and a dict respectively.

The same comment applies to the other location.

Reply:
Can I just directly set the args to be [] and {} in the parameter list?

Reply:
You could, but it's considered a bad practice because `[]` and `{}` are mutable objects, so if you have code like the following, it can give rise to unexpected behavior

```python
def f(x=[]):
    x.append(1)
    return x

f()  # returns [1]
f()  # returns [1, 1]
f()  # returns [1, 1, 1]
```

Reply:
Would the other remote function require this for args?

Reply:
Oh yeah, it looks like it does (could you also add a test for this?).

When I do `f._remote()`, I get

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rnishihara/ray_scratch/ray/python/ray/remote_function.py", line 111, in _remote
    kwargs)
  File "/Users/rnishihara/ray_scratch/ray/python/ray/signature.py", line 185, in extend_args
    args = list(args)
TypeError: 'NoneType' object is not iterable
```

Reply:
Good catch.

Reply:
How should I test this case?

Reply:
If you look this test 

https://github.com/ray-project/ray/blob/4c80177d6f3d402618b988badbe330d31816396a/python/ray/tests/test_basic.py#L830

I think it'd be enough to add a line like 
```python
assert ray.get(g._remote()) == [0]
```

that will ensure that we can call remote functions without specifying `args`.

Similarly you could add something to the same test like

```python
@ray.remote
class Actor2(object):
    def __init__(self):
        pass

    def method(self):
        pass

a = Actor2._remote()
ray.get(a.method._remote())
```

I think that will fail (before this PR), but after this PR it should succeed.